cms/djangoapps/contentstore/models.py

```diff
@@ -124,22 +124,22 @@ def upstream_version(self) -> int | None:
         Returns upstream block version number if available.
         """
         version_num = None
-        if hasattr(self.upstream_block, 'published'):
-            if hasattr(self.upstream_block.published, 'version'):
-                if hasattr(self.upstream_block.published.version, 'version_num'):
-                    version_num = self.upstream_block.published.version.version_num
+        if hasattr(self.upstream_block.publishable_entity, 'published'):
+            if hasattr(self.upstream_block.publishable_entity.published, 'version'):
+                if hasattr(self.upstream_block.publishable_entity.published.version, 'version_num'):
+                    version_num = self.upstream_block.publishable_entity.published.version.version_num
```

Comment:
FWIW, we should be able to simplify this down to use `openedx_learning.api.authoring.get_published_version`, like:

```python
published_version = get_published_version(self.upstream_block.id)
return published_version.version_num if published_version else None
```